### ISSUE TITLE
Add a common interface for JedisCommands and JedisClusterCommands

### ISSUE THREAD

**Issue by ptahchiev** [redis/jedis#1931]
Hey guys,
i'm having some trouble migrating this project:
https://github.com/jlinn/quartz-redis-jobstore
to jedis 3.0.x

So in the previous (2.9.x) version the `JedisCluster` was implementing `JedisCommands` and in 3.0.x it now implements a different interface `JedisClusterCommands` which has most of the `JedisCommands` methods in. Can we have all the common methods extracted to a common interface to both `JedisClusterCommands` and `JedisCommands`? 

Something like:
````
           TOP_INTF
     /               \
JedisCommands    JedisClusterCommands
````

**Comment by marcosnils** [comment]
I don't see why we cannot unify interfaces to avoid duplication. WDYT @gkorland / @sazzad16 ?

**Comment by ptahchiev** [comment]
Here is a PR: https://github.com/xetorthio/jedis/pull/1932

**Comment by ptahchiev** [comment]
Any update on this?

**Comment by ptahchiev** [comment]
Any update here? It blocks me to update to the latest spring-boot :(

**Comment by TristanPerry** [comment]
Another 👍 for this issue/PR - it would be quite useful to have.

**Comment by sazzad16** [comment]
Resolved by #2693
